examples/generative/corrdiff/datasets/cwb.py

```diff
@@ -347,97 +254,46 @@ def __init__(
         out_channels=(0, 17, 18, 19),
         img_shape_x=448,
         img_shape_y=448,
-        roll=False,
         add_grid=True,
         ds_factor=1,
         train=True,
         all_times=False,
-        n_history=0,
-        min_path=None,
-        max_path=None,
-        global_means_path=None,
-        global_stds_path=None,
-        normalization="v1",
     ):
+
+        self._dataset = dataset
         if not all_times:
-            self._dataset = (
-                FilterTime(dataset, is_not_2021)
+            self._time_indices = (
+                [i for i, t in enumerate(self._dataset.time()) if t.year != 2021]
```

Comment:
I wish we could avoid hard-coding the year here

Reply:
was hardcoded before, so this would be a "new feature". Would prefer to keep the new feature code separate from the refactors in this PR.